aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -296,6 +298,8 @@ impl AptosVM {
             }
             .map_err(|e| e.into_vm_status())?;
 
+            self.resolve_pending_code_publish(&mut session, gas_status)?;
```

Comment:
curious why this needs to called for every user txn. should this be only called when payload is module bundle.

Reply:
Module-bundle transactions will go away, and instead code is published via `code::publish_package`. Please also see the design doc.

Reply:
After reading [section](https://www.notion.so/aptoslabs/Releasing-Upgrading-Versioning-f9e50bd4e9394b598ca0803406c73d0c#be300c43db114a84a87bb01d59c98148), to deploy package,  I expect most of txn are not module publish TXN. would calling `resolve_pending_code_publish` lead to unnecessary step for other non-module publish TXN?  we can still call this `deploy_package` script function within the special module-bundle transaction type.

Reply:
The step this adds is `if let Some(x) == extract_request(session) ...`, so it is just a few instructions in Rust. How is that problematic?

This PR generally enables programmatic code publishing from *any* Move user code. This is unrelated to the module-bundle transaction type. Notice that this PR implements an alternative to what [@davidiw tried to land in the Move Repo](https://github.com/move-language/move/pull/285) but was rejected there.

Reply:
Ok, I see now. any user txn can publish code with this change.